SignalMessaging/environment/Environment.h

```diff
@@ -34,8 +34,9 @@
 @property (nonatomic, readonly) OWSMessageSender *messageSender;
 @property (nonatomic, readonly) OWSPreferences *preferences;
 
-+ (Environment *)current;
-+ (void)setCurrent:(Environment *)environment;
+@property (class, readonly, nonatomic) Environment *shared;
+
++ (void)setShared:(Environment *)environment;
```

Comment:
It seems like we'd get the same behavior by removing the `readonly` up a couple lines.

I'd argue doing so is less surprising to the reader since it's really **not** `readonly` - we've manually declared the exact same setter that would be generated by omitting `readonly`.